### ISSUE TITLE
callouts need to keep angle on log axis

### ISSUE THREAD

**Issue by lindsayplatt** [USGS-R/gsplot#205]
``` r
x <- runif(20, 0, 10)
y <- runif(20, 0, 50)

#callout with log is straight line
gs <- gsplot() %>%
         points(x,y, log='y')  %>%
         callouts(x[3], y[3], labels="stuff")
gs

#callout without log
gs <- gsplot() %>%
         points(x,y)  %>%
         callouts(x[3], y[3], labels="stuff")
gs
```

**Comment by lindsayplatt** [comment]
Unfortunately, the fix for getting limits to take callouts/error bars into account (moving the code into the ".gsplot" function) means that the values for par("usr") that are called in `callouts.gsplot` are the defaults of `[0,1,0,1]`. This is causing the callout lines to be very large with lower values and tiny with high values. 

``` r
# small values, large callouts
x <- runif(20, 0, 0.1)
y <- runif(20, 0, 0.5)
gs <- gsplot() %>%
    points(x,y)  %>%
    callouts(x[3], y[3], labels="stuff")
gs
```

![image](https://cloud.githubusercontent.com/assets/13220910/9394655/3b4ca302-474f-11e5-89d5-da97c0e7180a.png)

``` r
# large values, small callouts
x <- runif(20, 0, 100)
y <- runif(20, 0, 500)
gs <- gsplot() %>%
    points(x,y)  %>%
    callouts(x[3], y[3], labels="stuff")
gs
```

![image](https://cloud.githubusercontent.com/assets/13220910/9394673/5a3d75de-474f-11e5-8e01-a97b90bcb1f9.png)

**Comment by lindsayplatt** [comment]
The fix for the issue of angle, was just changing the second `if (par("xlog"))` (~line 51) to `if (par("ylog"))`. The issue of the length of the callout line segment is still an issue.

If you run the gsplot call twice, the par("usr") from the first plot remains and `callouts.R` uses that instead of the default, which produces the callout length wanted:

``` r
x <- runif(20, 0, 100)
y <- runif(20, 0, 500)

gs <- gsplot() %>%
    points(x,y)  %>%
    callouts(x[3], y[3], labels="stuff")
gs

gs <- gsplot() %>%
    points(x,y)  %>%
    callouts(x[3], y[3], labels="stuff")
gs
```

![image](https://cloud.githubusercontent.com/assets/13220910/9394810/298fef4c-4750-11e5-9c8d-08fcb0c92059.png)

**Comment by eread-usgs** [comment]
@lindsaycarr I am dictating for @jread-usgs  here because he's driving: even though I originally suggested that callouts should be included, i think this is actually a bad idea. Callouts dont actually include data, so they shouldnt shift the axes (I do however think error bars should modify the limits). The issues with callouts being off the page is somethign that I think should be fixed by going back to the original way they were calculated, but modifying the default argument for angle to be "auto". this would use the default angle of 30 degrees for all callouts with the exception of callouts that would end up off the figure, in these cases the angle would be switched to another opposite default that would flip them 180 degrees and modify the justification of the text. 

how does this sound to everyone?

**Comment by lindsayplatt** [comment]
Ok, I agree with callouts not being included in limits since they are not data. This would mean we should also change the default for arrows to "auto", which would address #210 that @mhines-usgs submitted.

**Comment by lindsayplatt** [comment]
fixed by #215
